Floofbot/Modules/Helpers/GameBoard.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Floofbot.Modules.Helpers
+{
+    public class GameBoard
+    {
+        private string[,] grid;
+        private int height;
+        private int width;
+        private int bombCount;
+
+        public GameBoard(int height, int width, int bombCount)
+        {
+            this.height = height;
+            this.width = width;
+            this.bombCount = bombCount;
+            grid = new string[this.height, this.width];
+            plantBombs();
+
+            for (int i = 0; i < this.height; i++)
+            {
+                for (int j = 0; j < this.width; j++)
+                {
+                    if (grid[i, j] != "||:bomb:||")
+                    {
+                        grid[i, j] = getBombCount(i, j).ToString();
+                    }
+                }
+            }
+        }
+
+        public int getBombCount(int x, int y)
+        {
+            int count = 0;
+
+            for (int i = -1; i <= 1; i++)
+            {
+                for (int j = -1; j <= 1; j++)
+                {
+                    if (!(i == 0 && j == 0))
+                    {
+                        //checks to see if the array goes out of bounds
+                        if (!(x + i < 0 || y + j < 0 || x + i >= grid.GetLength(0) || y + j >= grid.GetLength(1)))
+                        {
+                            if (grid[x + i, y + j] == "||:bomb:||")
```

Comment:
Not PR blocking: you can reform these two if statements as follows:
```cs
if (x + i >= 0 
    && y + j >= 0 
    && x + i < grid.GetLength(0) 
    && y + j < grid.GetLength(1) 
    && grid[x + i, y + j] == "||:bomb:||")
{
}
```
because the if statements "short-circuit", meaning if an earlier condition is false then it doesn't check the later conditions. Also, changing the or's to and's along with reversing the conditions is by De Morgan's Laws.